opts/network.go

```diff
@@ -0,0 +1,116 @@
+package opts
+
+import (
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+const (
+	networkOptName  = "name"
+	networkOptAlias = "alias"
+	driverOpt       = "driver-opt"
+)
+
+// NetworkAttachmentOpts represents the network options for endpoint creation
+type NetworkAttachmentOpts struct {
+	Target     string
+	Aliases    []string
+	DriverOpts map[string]string
+}
+
+// NetworkOpt represents a network config in swarm mode.
+type NetworkOpt struct {
+	options []NetworkAttachmentOpts
+}
+
+// Set networkopts value
+func (n *NetworkOpt) Set(value string) error {
+
+	longSyntax, err := regexp.MatchString(`\w+=\w+(,\w+=\w+)*`, value)
```

Comment:
It shouldn't be necessary to use a regexp here. You can apply the CSV parser to the short syntax as well, and you'll see a single field with no `=`. It would be something like:

```go
if len(fields) == 1 && strings.Contains(fields[0], "=") {
        // This is the short syntax
} else {
        // This is the long syntax
}
```

This is how `SecretOpt` supports long and short syntaxes.